### PR TITLE
[extractor/fuyin.tv] Add extractor

### LINKED ISSUE #2871: Support for fuyin.tv
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.02.04**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

_No response_

### Example URLs

-https://fuyin.tv/content/view/movid/2733/
-https://fuyin.tv/html/2733/44129.html

### Description

The first link is to a playlist of sorts. The blue buttons access the videos while the green ones download.
The second link is to an individual video that was accessed from clicking the blue button, specifically from episode one of the playlist.
YT-dlp download url for the second link is 
https://vod-hls.sanmanuela.com/07%E7%A6%8F%E9%9F%B3%E5%8A%A8%E6%BC%AB/%E5%85%89%E7%85%A7%E4%B8%AD%E5%8D%8E/01%E6%88%B4%E5%BE%B7%E7%94%9F%E7%9A%84%E6%95%85%E4%BA%8B.mp4/index.m3u8?_upt=df9141561645695278

### Verbose log

```shell
yt-dlp "https://fuyin.tv/html/52/20518.html" -v
[debug] Command-line config: ['https://fuyin.tv/html/52/20518.html', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Python version 3.9.0 (CPython 64bit) - macOS-12.2.1-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3.1, ffprobe 4.3.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://fuyin.tv/html/52/20518.html
[generic] 20518: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] 20518: Downloading webpage
[generic] 20518: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://fuyin.tv/html/52/20518.html
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1381, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1451, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 612, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/generic.py", line 3986, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://fuyin.tv/html/52/20518.html
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible.

This PR add site support for www.fuyin.tv,
Close #2871

### CODE REVIEW

File: yt_dlp/extractor/fuyintv.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        'url': 'https://www.fuyin.tv/html/2733/44129.html',
+        'info_dict': {
+            'id': '44129',
+            'ext': 'mp4',
+            'title': '第1集',
+        }
+    }]
+
+    def _real_extract(self, url):
+        playlist_id, video_id = self._match_valid_url(url).group('mov_id', 'url_id')
```
Comment (pukkandan) [inline yt_dlp/extractor/fuyintv.py:17]: `playlist_id` is unused?

File: yt_dlp/extractor/fuyintv.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from .common import InfoExtractor
+from ..utils import traverse_obj
+
+
+class FuyinTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)fuyin\.(?:tv)/(?:html)/(?P<mov_id>\w+)/(?P<url_id>\d+)(?:\.html)'
```
Comment (pukkandan) [inline yt_dlp/extractor/fuyintv.py:6]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?fuyin\.tv/html/(?P<mov_id>\w+)/(?P<url_id>\d+)\.html'
```
Why the unnecessary groups?

**Review by pukkandan — CHANGES_REQUESTED** [review]
We can merge this PR as-is if needed, but try to extract more metadata if you can

**Comment by pukkandan** [comment]
I future, pls make a new branch from `upstream/master` for each PR so that the unnecessary commits don't show

File: yt_dlp/extractor/fuyintv.py (hunk 1 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/fuyintv.py:17]: For now, it's unused, but i think can used later if set `--yes-playlist`

Comment (HobbyistDev) [reply yt_dlp/extractor/fuyintv.py:17]: Should i remove that ?

Comment (pukkandan) [reply yt_dlp/extractor/fuyintv.py:17]: --yes-playlist only applies when there is ambiguity whether a URL refers to video or playlist. It's not applicable here. If the group is unused, pls remove it

Comment (HobbyistDev) [reply yt_dlp/extractor/fuyintv.py]: Fixed in 3c495cbf546c7b489ebe597b34350d76c408d792

File: yt_dlp/extractor/fuyintv.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            'description': 'md5:21a3d238dc8d49608e1308e85044b9c3',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        json_data = self._download_json(
+            'https://www.fuyin.tv/api/api/tv.movie/url',
+            video_id, query={'urlid': f'{video_id}'})
+        webpage = self._download_webpage(url, video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/fuyintv.py:22]:
```suggestion
        webpage = self._download_webpage(url, video_id, fatal=False)
```

**Pull request merged**